Update to install.rst (#456)
opnsense/docs: source/manual/install.rst

------------

The **software setup** and installation of OPNsense® is available
for the `x86-64 <https://en.wikipedia.org/wiki/X86-64>`__ microprocessor
architecture only.

----------------
Embedded vs Full
----------------

> OPNsense offers two Image Types with all Major releases, Embedded (nano) Image 

Reviewer: I'd prefer lowercase when not referencing a trademark or otherwise wanting to highlight it. Serial, VGA, Nano and DVD work for me as image names. Makes it a little easier to process them.

```suggestion
OPNsense offers two Image types with all major releases: embedded and full images. [etc]
```

> OPNsense offers two Image types with all major releases: embedded and full images. [etc]
> and Full Images.  The Embedded Image is intended for environments where preinstalling 

Reviewer: looks like copy/paste error

```suggestion
OPNsense offers two Image types with all major releases: embedded and full images.
The Embedded Image is intended for environments where preinstalling 
```

the storage media is required due to a lack of local resources on the firewall 
like storage, and/or console access (VGA/Serial).  The image is tailored to reduce 
write cycles as well, but the image can be used anywhere.  Another reason for the 
Embedded Image is to eliminate the need for local console access for installing OPNsense.  
Installation is managed by prewriting the image to a storage device, installing the 
storage device, and booting the system.

Full Images provide installation tools like OPNsense Importer, Live Environment, 
and Installer.  Full Images are released to support different console/hardware installation 
requirements.  

Both image types can be installed and run from virtual disks (VM), `SD memory
cards <https://en.wikipedia.org/wiki/Secure_Digital>`__, 
USB disks, `solid-state
disks (SSD) <https://en.wikipedia.org/wiki/Solid-state_drive>`__, or `hard disk drives
(HDD) <https://en.wikipedia.org/wiki/Hard_disk_drive>`__.

> The main differences between an Embedded (nano) image and a Full images are:

Reviewer: Ideally we shouldn't confuse users with "nano" vs. embedded before the table below. Afterwards, "full" and "embedded" should no longer be references and the real image names should be used instead when required.

Author: So before the table, the image should be referred to as embedded or nano?

Author: After rereading your reply, we want to reference embedded before the table. Please let me know if I'm wrong.  I have already update.

Reviewer: simplify:

```suggestion
The main differences between embedded and full images are:
```

------------

The **software setup** and installation of OPNsense® is available
for the `x86-64 <https://en.wikipedia.org/wiki/X86-64>`__ microprocessor
architecture only.

----------------
Embedded vs Full
----------------

OPNsense offers two Image Types with all Major releases, Embedded (nano) Image 
and Full Images.  The Embedded Image is intended for environments where preinstalling 
the storage media is required due to a lack of local resources on the firewall 
like storage, and/or console access (VGA/Serial).  The image is tailored to reduce 
write cycles as well, but the image can be used anywhere.  Another reason for the 
Embedded Image is to eliminate the need for local console access for installing OPNsense.  
Installation is managed by prewriting the image to storage, installing the storage, and 
booting the system.

.. 

>    For nano images is console access required?

Reviewer 2: it's practical, but when written to a sd card and it boots normally, strictly you don't need it.

Embedded vs Full
----------------

> Since version 15.1.10 (04 May 2015) the option to install an

Author: Completely agree with the reference and kinda see your later point.   Merge the embedded/full with installation Media is the question?

> Since version 15.1.10 (04 May 2015) the option to install an

Reviewer: Merge first and second paragraph, yes. Trouble is the point it is trying to make is weak: embedded exist, full can be installed anywhere -- I don't have a better idea at the moment though.

> Since version 15.1.10 (04 May 2015) the option to install an

Author: I see your point on those two lines.  However, I didn't do any rewriting as I don't fully understand both lines.  Or should I say the contrast between both lines?  I just reorder the lines to match the Embedded vs Full, and the table below [ Embedded | Full ].   

Isn't the intended purpose of embedded more of the preinstall aspect?  "I'm going to download this image, write it to disk, install that disk into the firewall, and boot (no installation process per se).  Due to the nature of flash memory, do we run embedded versions in RAM drives?

> Since version 15.1.10 (04 May 2015) the option to install an

Reviewer: You are right, embedded image is a preinstalled image. That's the main difference. It's tailored to reduce write cycles as well, but can be used anywhere. Maybe that would be a good addition to these paragraphs.

+-----------------------+-----------------------+

> Embedded (nano) image store logging and cache data in memory only, while Full versions

Reviewer: ```suggestion
Embedded image store logging and cache data in memory only, while full versions
```

will keep the data stored on the local drive. A full version can mimic the
behavior of an embedded version by enabling RAM disks, this is especially
useful for SD memory card installations.

.. Warning::
    See the chapter :doc:`Hardware Sizing & Setup <hardware>` for further information 
    on hardware requirements prior to an install.

------------------
Installation Images
------------------

Depending on your hardware and use case, different installation options are available:

+--------+---------------------------------------------------+------------+
| Type   | Description                                       | Image Type |
+========+===================================================+============+
| dvd    | ISO image boots into a live environment in        | Full       |
|        | VGA-only mode with UEFI support                   |            |
+--------+---------------------------------------------------+------------+

> | vga    | USB image (.img) boots into a live environment    | Full       |

Reviewer: ```suggestion
| vga    | USB image boots into a live environment           | Full       |
```

+--------+---------------------------------------------------+------------+
| serial | USB image boots into live environment running in  | Full       |
|        | serial console (115200) mode only with            |            |
|        | UEFI support                                      |            |
+--------+---------------------------------------------------+------------+
| nano   | Image for preinstalling onto >=4 GB USB drives,   | Embedded   |
|        | SD, or CF cards for use with embedded devices     |            |
|        | running in serial console (115200) mode with      |            |
|        | secondary VGA support (no kernel messages though) |            |
+--------+---------------------------------------------------+------------+

.. Note::
   All Full Image types can run both **`OPNsense Importer <https://docs.opnsense.org/manual/install.html#opnsense-importer>`** 
   before booting into the Live environment and also run 
   **`Installer <https://docs.opnsense.org/manual/install.html#install-to-target-system>`** once booted into the Live environment.

.. Warning::
   Flash memory cards will only tolerate a limited number of writes and re-writes. For 

>    embedded (nano) versions memory disks for **/var/log** and **/tmp** are applied by 

Reviewer: ```suggestion
   Nano image memory disks for **/var/log** and **/tmp** are applied by 
```

   default to prolong CF (flash) card lifetimes.
   

>    To enable non-embedded versions: Go to :menuselection:`System --> Settings --> Miscellaneous --> Disk / Memory Settings`, 

Reviewer: ```suggestion
   To reverse this behaviour go to :menuselection:`System --> Settings --> Miscellaneous --> Disk / Memory Settings`, 
```

Author: This was on the original install page.  I believe who wrote was telling everyone you can make full image installations operation like nano images.

Reviewer: I know but I'm trying to get rid of "embedded" below the image table with the full/embedded column. From this point on "full" and "embedded" should be avoided.

Author: "To enable full versions" ?

Author: When you are referencing two different items.  In this case, Nano vs Full images.  We should capitalize the first letter.  I will take your lead.

Reviewer: well, yes but without either "embedded" or "full".

Reviewer: capitalizing breaks the reading flow for me. For Nano, VGA, Serial and DVD it makes sense, for everything else not.

Author: Again, I'll take your lead on this.  But breaking the flow of the reader is precisely the effect people need.  To know there is a difference between the two.  Remember, good man, you know this stuff WELL.  People reading this install guide have no clue.  They are still wrapping their minds around OPNsense.  ;)

    See the chapter :doc:`Hardware Setup <hardware>` for
    further information on hardware requirements prior to an install.

------------------
Installation Media
------------------

Depending on your hardware and use case, different installation options are available:

> +--------+---------------------------------------------------+------------+

Reviewer: nice addition, thanks!

+--------+---------------------------------------------------+------------+
| serial | USB image boots into live environment running in  | Full       |
|        | serial console (115200) mode only with            |            |
|        | UEFI support                                      |            |
+--------+---------------------------------------------------+------------+
| nano   | Image for preinstalling onto >=4 GB USB drives,   | Embedded   |
|        | SD, or CF cards for use with embedded devices     |            |
|        | running in serial console (115200) mode with      |            |
|        | secondary VGA support (no kernel messages though) |            |
+--------+---------------------------------------------------+------------+

.. Note::
   All Full Image types can run both **`OPNsense Importer <https://docs.opnsense.org/manual/install.html#opnsense-importer>`** 
   before booting into the Live environment and also run 
   **`Installer <https://docs.opnsense.org/manual/install.html#install-to-target-system>`** once booted into the Live environment.

.. Warning::

  Flash memory cards will only tolerate a limited number of writes

>   and re-writes. For embedded (nano) versions memory disks for **/var** and **/tmp** are

Reviewer: Nowadays this is:

```suggestion
  and re-writes. For embedded (nano) versions memory disks for **/var/log** and **/tmp** are
```

  on our release schedule is available from our package repository, see 
  `README <https://pkg.opnsense.org/releases/mirror/README>`.  You are encourage to updated 
  OPNsense after installation to be on the latest version available, see 
  `Update Page <https://docs.opnsense.org/manual/updates.html>`.

-------------------------
Download and Verification
-------------------------

The OPNsense distribution can be `downloaded <https://opnsense.org/download>`__
from one of our `mirrors <https://opnsense.org/download>`__.

OpenSSL is used for image file verification.  4 files are needed for verification process:

* The SHA-256 checksum file (<filename>.sha256)
* The bzip compressed Image file (<filename>.<image>.bz2)
* The signature file (<filename>.<image>.bz2.sig)
* The openssl public key (<filename>.pub)

Use one of the OPNsense mirrors to download these files:

1. Go to the bottom of OPNSense `download <https://opnsense.org/download>`__ page.
2. Click one of the available mirrors closest to your location.
3. Download one of each file mentioned above for your Image type.

--------------------

> OpenSSL and LibreSSL

Reviewer: nice catch ;)

   announcement, but 22.1.9 will not.

Once you download all the required files and verify that the public key matches 
the public key found in one of the alternate sources listed above, you can be relatively 
confident that the key has not been tampered with. To verify the downloaded image, run 
the following commands (substituting the filenames in brackets for the files you downloaded):

``openssl sha256 OPNsense-<filename>.bz2``

Match the checksum command output with the checksum vaules in file ``OPNsense-<version>-OpenSSL-checksums-amd64.sha256``.  
If the checksums don't match, redownload your image file.  If checksums match continue with the verification commands.

``openssl base64 -d -in OPNsense-<filename>.sig -out /tmp/image.sig``

``openssl dgst -sha256 -verify OPNsense-<filename>.pub -signature /tmp/image.sig OPNsense-<filename>.bz2``

If the output of the second command is “**Verified OK**”, your image file was verified 
successfully, and its safe to install from it. Any other outputs, and you may need 
to check your commands for errors, or the image file may have been compromised.

-------------------

> Installation Media

Reviewer: Is this a typo? We already had "media"?

Reviewer: Perhaps change this to "Installation media preparation" or similar to match next heading

Author: No, not in my suggestion to change the sections.  I renamed a few sections.  This was the first one.

Author: > Perhaps change this to "Installation media preparation" or similar to match next heading

Download and Verification are also parts of media preparation, which comes before Installation Media.  So I don't like that it.   But will take your lead if that's the case.

.. Note::

  **Please** be aware that the latest installation media does not always correspond 
  with the latest released version available. OPNsense installation images are provided 
  on a scheduled basis with major release versions in January and July. More information 
  on our release schedule is available from our package repository, see 

>   `README <https://pkg.opnsense.org/releases/mirror/README>`.  You are encourage to updated 

Reviewer 2: > encourage

encouraged ?

Author: I had this fixed, it reads: We are encouraged to update OPNsense after installation to be on the latest release available

to check your commands for errors, or the image file may have been compromised.

-------------------
Installation Media
-------------------

Now that you have downloaded and verified the installation image from above.  You must unpack the 
image file before you can write the image to disk.  For Unix-like OSes use ``bzip2 -d OPNsense-<filename>.bz2`` 
command.  For Windows use an application like `7zip <https://www.7-zip.org/download.html>`_.  The ``.bz2`` will 
be removed from the end of the filename after command/applcation completes.

After unpacking the image you can create the installation media. The easiest method to install 
OPNsense is to use USB "`vga <https://docs.opnsense.org/manual/install.html#installation-media>`_" 
Image. If your target platform has a serial console interface choose the 
“`serial <https://docs.opnsense.org/manual/install.html#installation-media>`_” image. If you 
need to know more about using the serial console interface, consult the :doc:`serial access how-to<how-tos/serial_access>`.

Write the image to a USB flash drive (>=1 GB) or hard disk, using either dd for Unix-like 

> OSes and for Windows use physdiskwrite or `Etcher <https://www.balena.io/etcher#download-etcher>`_.

Reviewer 2: quite some people seem to face challenges with `physdiskwrite` thanks for addressing this, maybe we could also add Rufus (https://rufus.ie/nl/) as an option, that seems to be the other popular option.

Author: Personally, I've only used phydiskwrite to confirm this install guide.  When using windows, I feel most users would stick to GUI-based tools.  Of course, I think Windows is adopting Linux tools.  Bash cli integration come win12 or 13, I bet.  haha  Should we remove phydiskwrite, and replace that with Etcher and Rufus?  

I will get Rufus added.  Should I get screenshots of both?

Author: @AdSchellevis remove physdiskwrite? Again I've never used it before myself.  Windows users are going to have much more success with Etcher and Rufus.

(either via keyboard and [virtual]monitor or :doc:`serial connectivity<how-tos/serial_access>`).  Next we need to know 
how to access the boot menu or the system bios (UEFI) to boot from the installation media.  Most times will be a function 
(F#), Del, or ESC key that needs to pressed immediately after powering on (or rebooting) the system.  Usually within the 
first 2 to 3 seconds from powering up.

.. Tip::

    OPNsense devices from the `OPNsense shop <https://shop.opnsense.com/>`__ use :code:`<ESC>` to enter the bios and boot selection
    options.

.. Note::

    Serial connectivity settings for DECXXXX devices can be found  :doc:`here </hardware/serial_connectivity>`

-------------------------
Installation Instructions
-------------------------

..
  Comment: Not sure how rubric:: are used.  I would like to replace Installation Instructions rubric with 
  section above.  I also don't know how :name: work

.. rubric:: Install Instructions
   :name: install-to-system

OPNsense installation boot process allows us to run several optional configuration steps. The 
boot process was designed to always boot into the live environment, allowing us to access the 

> GUI or even SSH directly. If a timeout was missed, restart the boot procedure.

Reviewer: ```suggestion
GUI or even SSH directly. If the importer timeout was missed, restart the boot procedure.
```

OPNsense Importer
-----------------

> All images have the OPNsense Importer feature that offers great flexibility in 

Reviewer: Technical docs should avoid opinions and exaggeration, "great" is one of those and can easily be dropped

Author: I completely agree.  It is an excellent feature if you ever have to recover a bad firewall.

> recovering failed firewalls quickly or testing new releases with existing 

Author: understood.

> configurations without installing. 

Reviewer: closer to the point: without overwriting the current installation by running the new version in memory with the existing configuration

Author: All images have the OPNsense Importer feature that offers flexibility in recovering failed firewalls or testing new releases without overwriting the current installation by running the new version in memory with the existing configuration.

Better?

Reviewer: yep

To use OPNsense Importer during the installation boot-up process do the following:

> 1. You must have a 2nd USB drive formatted with FAT or FAT32 File system.

Reviewer: The early importer can import from USB stick, but also just from the installed disk... (see point above)

Author: Oh, I see, I didn't realize that.  Just to be clear "Installed disk", means the existing drives from a previous installation?

Reviewer: yes, that's basically the idea of requiring a /conf directory to be present to import :)

   a. Preferable non-bootable USB drive.
2. Create a **conf** directory on the root of the USB drive

> 3. Put an unencrypted backup xml or configuration xml into /conf and rename the file to **config.xml**

Reviewer: backup and configuration xml are the same, perhaps shorten to "configuration"

Author: Well really two points here, you can use either a backup XML or a copy /conf/config.xml "configuration".  Another thing is you can encrypt backups, but you can't use encrypted .xml with the Importer.

Reviewer: yes, that's basically the idea of requiring a /conf directory to be present to import :)

Reviewer: it's a bit more complex: the backups are just copies of the authoritative config.xml from an older point in time. You can also import backups but these reside in /conf/backups directory and are only used when switching to them via GUI or console. The main config.xml is a configuration XML obtained somehow or edited otherwise and of course unencrypted as the decryption was not implemented for several reasons in this end of the system.

Author: Sorry you know the backend of the system more than me.  But from an end user, GUI-based user mostly, I know nothing of the /conf/backups/.  When I talk about backup.xml.  I'm referencing to the exported backups from the System > Configuration > Backups.  "Download configuration" button.  Place that file on a USB drive and use Importer in that context.  Maybe I should organize Importer into two functions, new installation using a backup.xml, and firewall recovery/testing new releases using existing /conf/config.xml on an installed drive?

Reviewer: well, my initial point about only using "configuration"... perhaps "previously exported configuration" or downloaded... just to be precise about the source and not the type of the configuration (backup vs. main)

Yes, first section for existing installs (shorter) and then second section for preparing a USB stick for import (e.g. new install where you don't have a previous install on the disk you want to retain)

  #. If unsuccessful, importer will error and return us to the device selection prompt. Suggest 
  repeating steps 1-3 again.

Live environment
----------------
After booting with an OPNsense Full Image (DVD, VGA, Serial), the firewall will 
be in the Live environment with and without the use of OPNsense Importer.  We 
can interact with the Live environment via Local Console, HTTPS GUI, or SSH.

By default, we can log into the shell using the user `root` with the password 
`opnsense` to operate the live environment via the local console.

The GUI is accessible at `https://192.168.1.1/ <https://192.168.1.1/>` using Username: 
`root` Password: `opnsense` by default (unless a previous configuration was imported).  

Using SSH we can access the firewall at IP `192.168.1.1`.  Both the `root` and `installer` 
users are available, using password `opnsense`. 

.. Note::
   That the installation media is read-only, which means your current live configuration will 
   be lost after reboot.

.. 
   Commect: I suggest we move Nano Image section either after "Install to target system" or to 
   before "System Boot Preparation".  Also needs a rewrite, I have it 40% done.  Does nano image 

>    use the 2nd interface as a default LAN interface using DHCP on 192.168.1.x network?

Reviewer 2: The interface configuration should be the same, the build system only "curries" it with memory/console settings (https://github.com/opnsense/tools/blob/fce1fe3fe006dba02be284e86430bf3d2ad9d892/config/23.1/extras.conf#L69-L73) if I'm not mistaken.

Moving the item is probably ok, from the text review it's a bit difficult to envision the final layout in my head.

Author: @AdSchellevis thank you for the reply.  But I'm not sure what I'm looking at in that link.  In my PR I have already moved the nano section and rewrote 80% of it.  I will boot up a nano image.  I believe the defaults are like full image defaults, LAN @ 192.168.1.1/24 on the first NIC, with DHCP pool, DNS, & WebGUI enabled. WAN on 2nd NIC w/DHCP IP.  

Thanks,

Author: So I have already moved the section and will boot up a nano image to see the defaults.  I will make future suggestions or have questions

Nano image
----------

.. 
   !rewrite - Using the Nano image for embedded systems, your firewall is already up and running.  
   The settings to minimize write cycles to relevant partitions by mounting these partitions in 
   system memory and reporting features are disabled by default.
.. 
   Other default settings like interfaces, DHCP, etc.  Or is a console required for nano images?
.. 
   "If there are?" What is the required configuration actions to start passing traffic? 

> 

Reviewer 2: see previous comment.

Author: Thank you and apology I haven't used the nano image before.  Do we think it's best to list out the default settings for the user to connect to the firewall after booting up?  I'm thinking this would be an extreme use case where the user flashes the image to the drive, installs, and boots without some type of console.
